packages/did-ethr-resolver/src/implementations/identityOwner.ts

```diff
@@ -0,0 +1,38 @@
+import { ethers } from 'ethers';
+import { EwJsonRpcSigner } from './ewJsonRcpSigner';
+import { EwPrivateKeySigner } from './ewPrivateKeySigner';
+import { EwSigner } from './ewSigner';
+
+/**
+* Extends EwSigner with a publicKey that is associated with the signer.
+* Intended to be instantiated from existing signers.
+*/
+export class IdentityOwner extends EwSigner {
+  /**
+  * @param signer an EwJsonRpcSigner
+  * @param publicKey Public key associated with signer.
+  * Isn't a prop of signer as signature is needed to derive.
+  * @returns IdentityOwner that uses the passed in signer
+  */
+  public static fromJsonRpcSigner(signer: EwJsonRpcSigner, publicKey: string, privateKey: string): IdentityOwner {
+
+    return new IdentityOwner(signer, publicKey, privateKey);
+  }
+
+  /**
+  * @param signer an EwPrivateKeySigner
+  * @returns IdentityOwner that uses the passed in signer
+  */
+  public static fromPrivateKeySigner(signer: EwPrivateKeySigner, privateKey: string): IdentityOwner {
+    const compressedPubKey = ethers.utils.computePublicKey(signer.publicKey, true)
+    return new IdentityOwner(signer, compressedPubKey.slice(2), privateKey);
+  }
+
+  private constructor(
+    ewsigner: EwSigner,
+    public readonly publicKey: string,
+    public readonly privateKey: string,
+  ) {
+    super(ewsigner.signer);
```

Comment:
This won't work as `JsonRpcSigner`, kinda by definition, doesn't expose the privateKey so I don't think we can expect that it can be passed in. You probably added the privateKey as it is used in other places (e.g. to instantiate Keys) but I think it needs to be an optional param. Does this make sense or do you have a different read of things?
```suggestion
  public static fromJsonRpcSigner(signer: EwJsonRpcSigner, publicKey: string): IdentityOwner {
    return new IdentityOwner(signer, publicKey);
  }

  /**
  * @param signer an EwPrivateKeySigner
  * @returns IdentityOwner that uses the passed in signer
  */
  public static fromPrivateKeySigner(signer: EwPrivateKeySigner): IdentityOwner {
    const compressedPubKey = ethers.utils.computePublicKey(signer.publicKey, true)
    return new IdentityOwner(signer, compressedPubKey.slice(2), signer.privateKey);
  }

  private constructor(
    ewsigner: EwSigner,
    public readonly publicKey: string,
    public readonly privateKey?: string,
  ) {
    super(ewsigner.signer);
```

Reply:
That makes totally sense indeed